### PR TITLE
Don't 404 on Boxes Search when there's No App

### PR DESCRIPTION
**Pull request by tzaffi**
After further discussions, align with `algod`'s behavior and don't 404 when searching for boxes on a non-existant application.

## References
https://github.com/algorand/go-algorand/issues/4575#issuecomment-1256205356

### CODE REVIEW

File: api/test_resources/boxes.json [hunk 1]
```diff
@@ -120,39 +120,19 @@
               "address": "OKUWMFFEKF4B4D7FRQYBVV3C2SNS54ZO4WZ2MJ3576UYKFDHM5P3AFMRWE",
               "amount": 999999499000,
               "amount-without-pending-rewards": 999999499000,
```
Comment (tzaffi) [inline api/test_resources/boxes.json:122]: With recent change, app 5 has been deleted at the time of querying.

File: api/test_resources/boxes.json [hunk 2]
```diff
@@ -303,28 +283,10 @@
                   "num-uint": 0
                 }
               }
-            },
```
Comment (tzaffi) [inline api/test_resources/boxes.json:306]: app 5 deleted now...

File: api/test_resources/boxes.json [hunk 3]
```diff
@@ -415,48 +377,25 @@
...
-              "local-state-schema": {
-                "num-byte-slice": 0,
-                "num-uint": 0
-              }
-            }
-          },
-          "current-round": 8
-        }
+        "statusCode": 404,
+        "body": "{\"message\":\"no application found for application-id: 5\"}\n"
```
Comment (tzaffi) [inline api/test_resources/boxes.json:395]: app 5 deleted now...

File: api/test_resources/boxes.json [hunk 4]
```diff
@@ -581,11 +520,17 @@
         "route": "/v2/applications/:application-id/boxes"
       },
       "response": {
-        "statusCode": 404,
-        "body": "{\"message\":\"no application found for application-id: round=8, appid=9223372036854775807\"}\n"
+        "statusCode": 200,
+        "body": "{\"application-id\":9223372036854775807,\"boxes\":[]}\n"
```
Comment (tzaffi) [inline api/test_resources/boxes.json:524]: No longer returning 404 when not finding boxes for an app, even when the app does not exist

**Review by winder — APPROVED** [review]

**Pull request merged**